cpp/src/plasma/eviction_policy.cc

```diff
@@ -28,13 +29,47 @@ void LRUCache::Add(const ObjectID& key, int64_t size) {
   // Note that it is important to use a list so the iterators stay valid.
   item_list_.emplace_front(key, size);
   item_map_.emplace(key, item_list_.begin());
+  used_capacity_ += size;
 }
 
 void LRUCache::Remove(const ObjectID& key) {
   auto it = item_map_.find(key);
   ARROW_CHECK(it != item_map_.end());
+  used_capacity_ -= it->second->second;
   item_list_.erase(it->second);
   item_map_.erase(it);
+  ARROW_CHECK(used_capacity_ >= 0) << DebugString();
+}
+
+void LRUCache::AdjustCapacity(int64_t delta) {
+  ARROW_LOG(INFO) << "adjusting global lru capacity from " << Capacity() << " to "
+                  << (Capacity() + delta) << " (max " << OriginalCapacity() << ")";
+  capacity_ += delta;
+  ARROW_CHECK(used_capacity_ >= 0) << DebugString();
+}
+
+int64_t LRUCache::Capacity() const { return capacity_; }
+
+int64_t LRUCache::OriginalCapacity() const { return original_capacity_; }
+
+int64_t LRUCache::RemainingCapacity() const { return capacity_ - used_capacity_; }
+
+void LRUCache::Foreach(std::function<void(ObjectID)> f) {
```

Comment:
`std::function<void(const ObjectID&)>` makes a bit more sense to me here but up to you.